fir/src/org/jetbrains/kotlin/idea/fir/applicators/AddArgumentNamesApplicators.kt

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2010-2021 JetBrains s.r.o. and Kotlin Programming Language contributors.
+ * Use of this source code is governed by the Apache 2.0 license that can be found in the license/LICENSE.txt file.
+ */
+package org.jetbrains.kotlin.idea.fir.applicators
+
+import com.intellij.psi.PsiComment
+import org.jetbrains.kotlin.idea.KotlinBundle
+import org.jetbrains.kotlin.idea.api.applicator.HLApplicatorInput
+import org.jetbrains.kotlin.idea.api.applicator.applicator
+import org.jetbrains.kotlin.name.Name
+import org.jetbrains.kotlin.psi.KtCallElement
+import org.jetbrains.kotlin.psi.KtLambdaArgument
+import org.jetbrains.kotlin.psi.KtPsiFactory
+import org.jetbrains.kotlin.psi.KtValueArgument
+import org.jetbrains.kotlin.psi.psiUtil.getPrevSiblingIgnoringWhitespace
+
+object AddArgumentNamesApplicators {
+    class SingleArgumentInput(val name: Name) : HLApplicatorInput
+
+    class MultipleArgumentsInput(val argumentNames: Map<KtValueArgument, Name>) : HLApplicatorInput
+
+    val singleArgumentApplicator = applicator<KtValueArgument, SingleArgumentInput> {
+        familyName(KotlinBundle.lazyMessage("add.name.to.argument"))
+
+        actionName { _, input -> KotlinBundle.message("add.0.to.argument", input.name)  }
+
+        isApplicableByPsi { element ->
+            // Not applicable when lambda is trailing lambda after argument list (e.g., `run {  }`); element is a KtLambdaArgument.
+            // Note: IS applicable when lambda is inside an argument list (e.g., `run({  })`); element is a KtValueArgument in this case.
+            !element.isNamed() && element !is KtLambdaArgument
+        }
+
+        applyTo { element, input ->
+            val expression = element.getArgumentExpression() ?: return@applyTo
+            val name = input.name
+
+            val prevSibling = element.getPrevSiblingIgnoringWhitespace()
+            if (prevSibling is PsiComment && """/\*\s*$name\s*=\s*\*/""".toRegex().matches(prevSibling.text)) {
+                prevSibling.delete()
+            }
+
+            val newArgument = KtPsiFactory(element).createArgument(expression, name, element.getSpreadElement() != null)
+            element.replace(newArgument)
+        }
+    }
+
+    val multipleArgumentsApplicator = applicator<KtCallElement, MultipleArgumentsInput> {
+        familyAndActionName(KotlinBundle.lazyMessage("add.names.to.call.arguments"))
+
+        isApplicableByPsi { element ->
+            // Note: `KtCallElement.valueArgumentList` only includes arguments inside parentheses; it doesn't include a trailing lambda.
```

Comment:
Thank you for leaving these comments, good to learn!

But, just curious. Then, where is that trailing lambda stored in `KtCallElement` ?

Reply:
A call like `foo(1, 2) {}` looks something like this:
```
KtCallExpression (a subclass of KtCallElement)  // `foo(1, 2) {}`
 ↳ KtValueArgumentList  // `(1, 2)`
    ↳ KtValueArgument  // `1`
    ↳ KtValueArgument  // `2`
 ↳ KtLambdaArgument  // `{}`
```